core/types/transaction.go

```diff
@@ -483,6 +483,21 @@ func (tx *Transaction) SetCodeAuthorizations() []SetCodeAuthorization {
 	return setcodetx.AuthList
 }
 
+// Authorities returns a list of each authorization's corresponding authority.
+func (tx *Transaction) Authorities() []common.Address {
+	setcodetx, ok := tx.inner.(*SetCodeTx)
+	if !ok {
+		return nil
+	}
+	auths := make([]common.Address, len(setcodetx.AuthList))
```

Comment:
Shouldn't this be `auths := make([]common.Address, 0, len(setcodetx.AuthList))` ? The current code appends addresses after a list of null addresses, it does not seem to be correct.

Reply:
yep you're right!